pkcs11-sys/build.rs

```diff
@@ -1,8 +1,24 @@
-use std::env;
-use std::path::PathBuf;
-
 fn main() {
-    // Generate bindings.
+    #[cfg(feature = "generate-bindings")]
+    {
+        generate_bindings();
+    }
+
+    #[cfg(not(feature = "generate-bindings"))]
+    {
+        let supported_platforms = vec![String::from("x86_64-unknown-linux-gnu")];
+        let target = std::env::var("TARGET").unwrap();
+
+        // check if target is in the list of supported ones or panic with nice message
+        if !supported_platforms.contains(&target) {
+            panic!(format!("Compilation target ({}) is not part of the supported targets ({:?}). Please compile with the \"generate-bindings\" feature or add support for your platform :)", target, supported_platforms));
+        }
+    }
```

Comment:
You can change the two `#[cfg...]` stuff with `if cfg!() { .. } else { .. }`

Reply:
Unfortunately not because when using the `cfg!` macro you need the things inside them to be available at compile time (like bindgen) whereas with the attribute `#[cfg...]` they are "removed" from compilation

Reply:
Ah, makes sense!